launch_testing/launch_testing/__init__.py

```diff
@@ -41,7 +41,7 @@ def __init__(
         self.stdout_data = io.BytesIO()
         self.stderr_data = io.BytesIO()
         self.regex_match = regex_match
-        self.matched = False
+        self.matched = len(self.expected_lines) == 0
```

Comment:
Why should a test use this fancy `InMemoryHandler` if it doesn't specify any expected lines? I would think the launch file should not specify the handler in this case instead.